### ISSUE TITLE
Search/Lookup an Asset by Custom Field via API

### ISSUE THREAD

**Issue by athc0** [grokability/snipe-it#9576]
**Server (please complete the following information):**
 - Snipe-IT Version 5.1.2-pre build 5830 (develop)
 - OS/Web/PHP: Hosted by Snipe-IT

**Is your feature request related to a problem? Please describe.**
I'd like to be able to search for or lookup an asset by a custom field. Our devices have multiple unique identifying pieces of information and I need to be able to find one based on any number of values. I've tried with Postman with no luck.

**Describe the solution you'd like**
Basically make something like the below work. We may store some unique identifying data about a device in the notes of another device, which makes search difficult to use as it sometimes returns multiple rows.

Example 1:
https://snipe-it.io/api/v1/hardware?search=1234567890&field=_snipeit_customfield_12

Example 2:
https://snipe-it.io/api/v1/hardware?_snipeit_customfield_12=1234567890

**Describe alternatives you've considered**
Excel and import, gross.

**Comment by leitwerk-ag** [comment]
That would be a very useful feature!

**Comment by snipe** [comment]
You can already search by custom field values. It’s written into the search scope, which is why you can search for them using the asset search at the top of the table. https://github.com/snipe/snipe-it/blob/db0a078c0b585956a24b0594ee47606136ef638a/app/Http/Controllers/Api/AssetsController.php#L113

**Comment by leitwerk-ag** [comment]
thanks for the super-fast reply! :)
Not sure, I understand correctly:
did you mean, that I can already full-text search the assets for all fields including the custom fields? I'm aware of that; this gives me lots of false positive matches, though for matches in other [custom] fields. So this is actually how I use it today - the results are not very precise, though. 
Maybe if it was possible to prepend the custom field label to limit the search to this field? 
Like so: (with "stickers" being the label of the custom field and "some" being the text to search for in that field)
stickers=some
That would already help a lot. Would that simplify the implementation?

**Comment by athc0** [comment]
Perhaps I'm missing something. Are you saying that it's possible to search by a specific field or that search already includes all fields?

I suggested this solution because often I perform a search via the API and unrelated assets will be returned because someone saved the data that was being searched in notes of an asset or some other unrelated field, which then need to be filtered client-side. Search that allows specifying individual fields would be less load on the DB server and return more specific results.

**Comment by athc0** [comment]
@snipe Thank you!
